app/ts/MigrationUI/components/Migration.tsx

```diff
@@ -0,0 +1,113 @@
+	}
+
+	const approveRepV1ForMigration = async () => {
+		const account = maybeAccountAddress.peek()
+		if (account === undefined) throw new Error('missing maybeAccountAddress')
+		if (v1ReputationBalance.deepValue === undefined) throw new Error('missing v1ReputationBalance balance')
+		await approveErc20Token(account.value, REPUTATION_V1_TOKEN_ADDRESS, REPV2_TOKEN_ADDRESS, v1ReputationBalance.deepValue)
+	}
+
+	if (universe.deepValue === undefined || reputationTokenAddress.deepValue === undefined || universeForkingInformation.deepValue === undefined) return <></>
+	return <div class = 'subApplication'>
+		<button class = 'button is-primary' onClick = { update }>Update data</button>
+		<div class = 'panel'>
+			<div style = 'display: grid'>
+				<span><b>Universe Name:</b>{ getUniverseName(universe.deepValue) }</span>
+				<span><b>Universe Address:</b>{ universe.deepValue }</span>
+				<span><b>Reputation V2 Address For The Universe:</b>{ universe.deepValue }</span>
+				<span><b>Is Universe Forking:</b>{ universeForkingInformation.deepValue.isForking ? 'Yes' : 'No' }</span>
+				<span><b>Forking End Time:</b>{ universeForkingInformation.deepValue.forkEndTime === undefined ? 'Not Forking' : formatUnixTimestampISO(universeForkingInformation.deepValue.forkEndTime) }</span>
+				<span><b>Has Forking Time Ended:</b>{ universeForkingInformation.deepValue.forkEndTime !== undefined && universeForkingInformation.deepValue.forkEndTime < new Date().getUTCSeconds() ? 'Yes' : 'No' }</span>
+				<span><b>Forking Market:</b>{ universeForkingInformation.deepValue.forkingMarket === undefined ? 'No Forking Market' : universeForkingInformation.deepValue.forkingMarket }</span>
+				<span><b>Your Reputation V2 Balance:</b>{ v2ReputationBalance.deepValue !== undefined ? `${ bigintToDecimalString(v2ReputationBalance.deepValue, 18n) } REP` : '' }</span>
+			</div>
+		</div>
+		{ universeForkingInformation.deepValue.isForking ? <>
+			<div class = 'panel'>
+				<Market marketData = { forkingMarketData }/>
+				<MarketReportingWithoutStake outcomeStakes = { forkingoutcomeStakes } selectedOutcome = { selectedOutcome }/>
+			</div>
+			<button class = 'button is-primary' onClick = { migrateReputationToChildUniverseByPayoutButton }>Migrate Reputation to the new universe</button>
+		</> : <></> }
+		{ isGenesisUniverseField ? <>
+			<div class = 'panel'>
+				<div style = 'display: grid'>
+					<span><b>Reputation V1 Address:</b>{ REPUTATION_V1_TOKEN_ADDRESS }</span>
+					<span><b>Your Reputation V1 Balance:</b>{ v1ReputationBalance.deepValue !== undefined ? `${ bigintToDecimalString(v1ReputationBalance.deepValue, 18n) } REPv1` : '' }</span>
+				</div>
+			</div>
+			<button class = 'button is-primary' onClick = { approveRepV1ForMigration }>Approve Reputation V1 For Migration</button>
+			<button class = 'button is-primary' onClick = { migrateFromRepV1toRepV2GenesisTokenButton }>Migrate Reputation V1 Tokens To Reputation V2</button>
```

Comment:
We should check to see if they have approval already, if not hide the button.

Does migration work with a partial approval, or will the contract revert?  If it will revert then we should hide this button and only show button to approve the rest.  If the call will just do a partial migration then we should probably add some text indicating that.

Reply:
You cannot pick how much to migrate and it fails if you have not approved all that you have when sending the transaction